frontend/javascripts/admin/job/job_hooks.ts

```diff
@@ -0,0 +1,107 @@
+import features from "features";
+import { getJobs } from "admin/admin_rest_api";
+import { useInterval } from "beautiful-react-hooks";
+import Toast from "libs/toast";
+import { OxalisState } from "oxalis/store";
+import { useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+import { APIJob } from "types/api_flow_types";
+
+export function useStartAndPollJob({
```

Comment:
Good point! I created a new issue: https://github.com/scalableminds/webknossos/issues/6599
I think, refactoring existing code might not be worth it (especially, since the hook cannot be used in class-components), but especially for new job-integrations, we should keep this in mind :)